src/intelligence_layer/connectors/studio/studio.py

```diff
@@ -191,3 +222,53 @@ def _upload_trace(self, trace: ExportedSpanList) -> str:
             case _:
                 response.raise_for_status()
         return str(response.json())
+
+    def submit_dataset(
+        self,
+        dataset: StudioDataset,
+        examples: Iterable[StudioExample[Input, ExpectedOutput]],
+    ) -> str:
+        """Submits a dataset to Studio.
+
+        Args:
+            dataset: Dataset to be uploaded
+            examples: Examples of Dataset
+
+        Returns:
+            id of created dataset
+        """
+        url = urljoin(self.url, f"/api/projects/{self.project_id}/datasets")
+        source_data_list = [
+            example.model_dump_json()
+            for example in sorted(examples, key=lambda x: x.id)
+        ]
+
+        source_data_file = "\n".join(source_data_list).encode()
+
+        data = {
+            "name": dataset.name,
+            "labels": list(dataset.labels) if dataset.labels is not None else [],
```

Comment:
why do we wrap it into a list here, is labels not a list already?

Reply:
Labels are of type set